packages/core/lib/commands/compile.js

```diff
@@ -61,17 +61,14 @@ const command = {
       }
     ]
   },
-  run: function (options, done) {
+  run: async function (options) {
     const TruffleError = require("@truffle/error");
     const WorkflowCompile = require("@truffle/workflow-compile");
     const Config = require("@truffle/config");
     const config = Config.detect(options);
 
     if (config.list !== undefined) {
-      command
-        .listVersions(config)
-        .then(() => done())
-        .catch(done);
+      return command.listVersions(config);
     } else {
```

Comment:
This else clause can be removed because of the return above.
```
    if (config.list !== undefined) {
      return command.listVersions(config);
    } else {
      if (...) {}
    }

```
can become
```
    if (config.list !== undefined) {
      return command.listVersions(config);
    } 
    if (...) {}

```

Reply:
You can take the pythonista out of the Python... :D